src/providers/pkcs11_provider/key_management.rs

```diff
@@ -280,7 +286,15 @@ impl Pkcs11Provider {
         let exponent_object = &public_key.public_exponent.as_unsigned_bytes_be();
         let bits = key_attributes.bits;
         if bits != 0 && modulus_object.len() * 8 != bits {
-            error!("If the bits field is non-zero (value is {}) it must be equal to the size of the key in data.", key_attributes.bits);
+            if crate::utils::GlobalConfig::log_error_details() {
+                error!(
+                    "`bits` field of key attributes (value: {}) was not equal to the size of the key in `data` (value: {}).",
+                    key_attributes.bits,
+                    modulus_object.len() * 8
+                );
+            } else {
+                error!("`bits` field of key attributes was not equal to the size of the key in `data`.");
+            }
```

Comment:
I think we lose the information of it being possibly zero

Reply:
? if `bits` is 0 then it doesn't go into this branch of the if on line 282/288

Reply:
Or do you mean if `modulus_object.len()` is 0?

Reply:
I just meant that before this message might help the developper giving him the information that he can either leave this field to 0 or he has to put the correct size. Currently it only seems like he *has* to put the exact size.